src/main/java/alfred/Alfred.java

```diff
@@ -0,0 +1,116 @@
+    private Storage storage;
+
+    /**
+     * Initializes a new instance of <code>Alfred</code> with the specified file path
+     * for saving and loading tasks. Attempts to load tasks from storage and handles
+     * errors related to loading and file corruption.
+     *
+     * @param filePath The path to the file used for storing task data.
+     */
+    public Alfred(String filePath) {
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.loadTasks());
+        } catch (IOException e) {
+            Ui.showLoadingError(e);
+            this.tasks = new TaskList();
+        } catch (AlfredException e) {
+            Ui.showCorruptedSaveError(e);
+            storage.clearStorage();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts the Alfred application, managing user interactions and
+     * executing commands until the user exits the application.
+     * Commands include listing tasks, marking tasks as done/undone,
+     * deleting tasks, and adding new tasks.
+     */
+    public void run() {
+        // Greet user
+        Ui.greet();
+
+        // Create a Scanner Object
+        Scanner in = new Scanner(System.in);
+
+        String input = in.nextLine();
+        while (!input.equals("bye")) {
+            String command = Parser.getCommand(input);
+
+            switch (command) {
+            case "list":
+                Ui.showList(tasks.getTasks());
+                break;
+            case "mark":
+            case "unmark":
+                boolean isMark = command.equals("mark");
+                if (Parser.isValidCommand(input, command, tasks.getTasksCount())) {
+                    int taskNumber = Parser.getTaskNumberFromInput(input);
+                    tasks.updateTaskStatus(taskNumber, isMark);
+                }
+                break;
+            case "delete":
+                if (Parser.isValidCommand(input, "delete", tasks.getTasksCount())) {
+                    int taskNumber = Parser.getTaskNumberFromInput(input);
+                    tasks.deleteTask(taskNumber);
+                }
+                break;
+            case "find":
+                String keyword = Parser.getKeyword(input);
+                List<Task> foundTasks = tasks.findTasks(keyword);
+                Ui.showFoundTasks(foundTasks);
+                break;
+            default:
+                if (Task.isCreateTaskCommand(input)) {
+                    try {
+                        Task task = Task.initialise(input);
+                        tasks.addTask(task);
+                        Ui.showAddedTaskMessage(task, tasks.getTasksCount());
+                    } catch (AlfredException e) {
+                        Ui.showAlfredError(e);
+                    } catch (Exception e) {
+                        Ui.showUnexpectedError(e);
+                    }
+                } else {
+                    Ui.showInvalidCommand();
+                }
```

Comment:
Perhaps it would be better to extract this logic into another method to avoid deep nesting? 😄 
```
private void handleDefault() {
   --logic here-
}
```

Reply:
I think that is a great suggestion, thanks! 😄